python_modules/dagster/dagster/core/asset_defs/assets.py

```diff
@@ -20,7 +20,10 @@ def __init__(
         partitions_def: Optional[PartitionsDefinition] = None,
         partition_mappings: Optional[Mapping[AssetKey, PartitionMapping]] = None,
         asset_deps: Optional[Mapping[AssetKey, AbstractSet[AssetKey]]] = None,
-        # if adding new fields, make sure to handle them in both with_replaced_asset_keys
+        can_subset: bool = False,
+        unconnected_inputs: AbstractSet[str] = None,
```

Comment:
I might not be fully grokking what's going on here yet, but seeing unconnected inputs here strikes me as a little bit odd, because I think of unconnected inputs as a job-level concept. Would it make sense to call this something like `unused_inputs`?

Reply:
Also, does this need to be passed in? Can it not be inferred from the intra-op deps?

Reply:
yeah unused is a much better name

Reply:
the "unconnected" bit was referencing that they're "inputs that will be unconnected"

Reply:
actually just refactored this to only keep track of the selected asset keys that you want to produce